app/src/main/java/com/asfoundation/wallet/App.kt

```diff
@@ -0,0 +1,163 @@
+import com.crashlytics.android.core.CrashlyticsCore
+import com.flurry.android.FlurryAgent
+import dagger.android.AndroidInjector
+import dagger.android.DispatchingAndroidInjector
+import dagger.android.HasAndroidInjector
+import io.fabric.sdk.android.Fabric
+import io.intercom.android.sdk.Intercom
+import io.reactivex.exceptions.UndeliverableException
+import io.reactivex.plugins.RxJavaPlugins
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class App : MultiDexApplication(), HasAndroidInjector, BillingDependenciesProvider {
+  @Inject
+  lateinit var androidInjector: DispatchingAndroidInjector<Any>
+  @Inject
+  lateinit var proofOfAttentionService: ProofOfAttentionService
+  @Inject
+  lateinit var inAppPurchaseInteractor: InAppPurchaseInteractor
+  @Inject
+  lateinit var appcoinsOperationsDataSaver: AppcoinsOperationsDataSaver
+  @Inject
+  lateinit var bdsApi: BdsApi
+  @Inject
+  lateinit var walletService: WalletService
+  @Inject
+  lateinit var proxyService: ProxyService
+  @Inject
+  lateinit var appcoinsRewards: AppcoinsRewards
+  @Inject
+  lateinit var billingMessagesMapper: BillingMessagesMapper
+  @Inject
+  lateinit var bdsapiSecondary: BdsApiSecondary
+  @Inject
+  lateinit var idsRepository: IdsRepository
+  @Inject
+  lateinit var logger: Logger
+  @Inject
+  lateinit var rakamAnalytics: RakamAnalytics
+
+  override fun onCreate() {
+    super.onCreate()
+    val appComponent = DaggerAppComponent.builder()
+        .application(this)
+        .build()
+    appComponent.inject(this)
+    setupRxJava()
+    setupWorkManager(appComponent)
+    setupSupportNotificationWorker()
+    if (!BuildConfig.DEBUG) {
+      FlurryAgent.Builder()
+          .withLogEnabled(false)
+          .build(this, BuildConfig.FLURRY_APK_KEY)
+      logger.addReceiver(FlurryReceiver())
+    }
+    Fabric.with(this, Crashlytics.Builder().core(
+        CrashlyticsCore.Builder().disabled(BuildConfig.DEBUG)
+            .build())
+        .build())
+    logger.addReceiver(CrashlyticsReceiver())
+    inAppPurchaseInteractor.start()
+    proofOfAttentionService.start()
+    appcoinsOperationsDataSaver.start()
+    appcoinsRewards.start()
+    Intercom.initialize(this, BuildConfig.INTERCOM_API_KEY,
+        BuildConfig.INTERCOM_APP_ID)
+    Intercom.client()
+        .setInAppMessageVisibility(Intercom.Visibility.GONE)
+    rakamAnalytics.start()
+  }
+
+  private fun setupRxJava() {
+    RxJavaPlugins.setErrorHandler { throwable: Throwable ->
+      if (throwable is UndeliverableException) {
+        if (BuildConfig.DEBUG) {
+          throwable.printStackTrace()
+        } else {
+          FlurryAgent.onError("ID", throwable.message!!, throwable)
+        }
+      } else {
+        throw RuntimeException(throwable)
+      }
+    }
+  }
+
+  private fun setupWorkManager(appComponent: AppComponent) {
+    WorkManager.initialize(this,
+        Configuration.Builder().setWorkerFactory(appComponent.daggerWorkerFactory())
```

Comment:
Improve kotlin codestyle by backspacing before Configuration.Builder() and applying codestyle again

Reply:
Done